shared/folders/files/paperkey.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+import {navigateUp} from '../../actions/router'
+import {checkPaperKey} from '../../actions/unlock-folders'
+import HiddenString from '../../util/hidden-string'
+import Render from '../../login/register/paper-key/index.render'
+
+type Props = $Shape<{
+  waiting: boolean,
+  onBack: () => void,
+  checkPaperKey: (paperKey: HiddenString) => void,
+}>
+
+type State = {
+  paperKey: string,
+  submitted: boolean,
+}
+
+class PaperKey extends Component<void, Props, State> {
+  state: State;
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      paperKey: '',
+      submitted: false,
+    }
+  }
+
+  componentWillReceiveProps (nextProps) {
+    if (!nextProps.waiting && this.state.submitted) {
```

Comment:
if we submitted and we're no longer waiting we just nav back. we'll have to test this further